smartapp-core/src/main/java/com/smartthings/sdk/smartapp/core/extensions/contextstore/DefaultInstalledAppContext.java

```diff
@@ -0,0 +1,148 @@
+package com.smartthings.sdk.smartapp.core.extensions.contextstore;
+
+import java.time.Instant;
+
+import com.smartthings.sdk.smartapp.core.models.InstallData;
+import com.smartthings.sdk.smartapp.core.models.InstalledApp;
+import com.smartthings.sdk.smartapp.core.models.UpdateData;
+
+
+/**
+ * This is the default implementation of InstalledAppContext which stores
+ * basic information related to installed apps. In particular, this includes
+ * token information to allow for keeping them up-to-date for out of band
+ * API requests.
+ */
+public class DefaultInstalledAppContext implements InstalledAppContext {
+    private InstalledApp installedApp;
+    private String authToken;
+    private Instant authTokenExpiration;
+    private String refreshToken;
+    private Instant refreshTokenExpiration;
+
+    public static DefaultInstalledAppContext from(InstallData installData) {
+        return new DefaultInstalledAppContext(installData.getInstalledApp(), installData.getAuthToken(),
+                installData.getRefreshToken());
+    }
+
+    public static DefaultInstalledAppContext from(UpdateData updateData) {
+        return new DefaultInstalledAppContext(updateData.getInstalledApp(), updateData.getAuthToken(),
+                updateData.getRefreshToken());
+    }
+
+    public DefaultInstalledAppContext() {
+        // Include empty constructor to make this class as amenable to
+        // persistence engines as possible.
+    }
+
+    public DefaultInstalledAppContext(InstalledApp installedApp, String authToken, String refreshToken) {
+        this.installedApp = installedApp;
+        this.authToken = authToken;
+        this.refreshToken = refreshToken;
+
+        authTokenExpiration = Instant.now();
+        refreshTokenExpiration = authTokenExpiration;
```

Comment:
Aren't Auth Token and Refresh Token expirations different?

Typically an auth token is shorter lived and a refresh token is longer lived.

Reply:
I was just treating them as expired as they come in for my first run and meant to update this a bit but forgot. I'll fix this and probably add something to configure it in case it changes.

Reply:
Ok, this should be fixed now.